### ISSUE TITLE
[Feature] peerDependencies

### ISSUE THREAD

**Issue by anilanar** [lerna/lerna#375]
I have the following setup:

```
root
| -- packages
     | -- foo
     | -- bar
     | -- quu
```

Both `foo` and `bar` have a peer dependency to `quu`. The reason is `quu` has global state and to be able to force npm to use a single instance of package `quu`, using peer dependency mechanism is a must. 

`foo` has a dependency to `bar`.

To be able to run unit tests in `foo`, `quu` is included as a `devDependency`, because `peerDependencies` are not installed (which is fine).

**Problem**: When running tests in `foo`, `foo` calls some code from `bar`, `bar` calls some code from `quu` but because `bar` doesn't have `quu` in its node_modules (remember? it's a peer dependency), I cannot run my tests unless I add `quu` to `bar` as a dev dependency which is not fine.

Adding `quu` to top level package.json is not an option because top-level dependencies are not linked.

What would you suggest to handle this case?

This issue might be a counter-argument to #160 and #318.

**Comment by thejameskyle** [comment]
Yarn will hopefully figure out the peerDependencies problem that npm currently has and when we switch to Yarn this should be part of that.

**Comment by gre** [comment]
I wanted to port my project https://github.com/gre/gl-react to using lerna but also reached this problem. And I'm afraid switching to *yarn* does not seem to fix it.

This bug basically show up with these 3 packages I have:
- cookbook: dep to gl-react & gl-react-dom
- gl-react: no deps
- gl-react-dom: **peer** deps to gl-react

After a `lerna bootstrap`, cookbook correctly have node_modules/gl-react and node_modules/gl-react-dom but it's something that breaks from when resolving gl-react from inside gl-react-dom ( `Module not found: 'gl-react' in $PROJROOT/packages/gl-react-dom/lib` )

so basically, yeah there is no node_modules/gl-react-dom/node_modules/gl-react because it's defined as a peer dep, by why isn't solved via node_modules/gl-react-dom ?

(cookbook is a create-react-app bootstrapped project)

I'm not sure what's a workaround but i'll probably have to not use lerna until it's fixed.

**Comment by mfellner** [comment]
@gre Have you tried hoisting everything? By default --hoist should add all dependencies to the root thus making it available to all packages (https://github.com/lerna/lerna#--hoist-glob).

**Comment by gre** [comment]
oh great! will try it shortly

**Comment by gre** [comment]
it does not seem to help my bug :'(

**Comment by wtgtybhertgeghgtwtg** [comment]
> I cannot run my tests unless I add quu to bar as a dev dependency which is not fine.

More than that; as of the latest release, it just seems to hang.

**Comment by evocateur** [comment]
If you need to test `peerDependencies`, they need to be specified as `devDependencies` too. That's just how things work with the weird beast that is `peerDependencies`.
